dapps/marketplace/src/pages/create-listing/listing-types/FractionalListing/FractionalListing.js

```diff
@@ -4,25 +4,42 @@ import Steps from '../Steps'
 import TitleDescription from '../../TitleDescription'
 import Pricing from './Pricing'
 import Availability from './Availability'
+import Location from './Location'
+import LocationObfuscation from './LocationObfuscation'
 import Images from '../../Images'
 import Review from './Review'
 
-const FractionalListing = props => (
-  <Steps
-    {...props}
-    steps={[
-      { step: 1, component: TitleDescription, require: 'subCategory' },
-      { step: 2, component: Pricing, path: 'pricing', require: 'quantity' },
-      {
-        step: 3,
-        component: Availability,
-        path: 'availability',
-        require: 'title'
-      },
-      { step: 4, component: Images, path: 'images', require: 'price' },
-      { step: 5, component: Review, path: 'review' }
-    ]}
-  />
-)
+const FractionalListing = props => {
+  const isHausing = props.listing.subCategory === 'schema.housing'
+
+  const steps = [
+    { step: 1, component: TitleDescription, require: 'subCategory' },
+    { step: 2, component: Pricing, path: 'pricing', require: 'quantity' },
+    {
+      step: 3,
+      component: Availability,
+      path: 'availability',
+      require: 'title'
+    }
+  ]
+  if (isHausing) {
+    steps.push({ step: 4, component: Location, path: 'location' })
+    steps.push({
+      step: 5,
+      component: LocationObfuscation,
+      path: 'locationObfuscate'
+    })
+  }
+
+  steps.push({
+    step: isHausing ? 6 : 4,
```

Comment:
nit: Instead of the conditional check, you could perhaps have a variable `counter` that starts with 1 and replace the step values with `counter++` in all steps. Would be less confusing.

Reply:
Yup agreed, your idea is much cleaner